### DISCUSSION TITLE
FluentDataGrid with EntityFramework and large data set

### DISCUSSION THREAD

**Discussion by domenicocanavero** [microsoft/fluentui-blazor#1129] [Q&A] [answered]
I need a sample with a FluentDataGrid bind to a linq query on a EFs dbcontext, with a large data set and then paging and all optimizations to improve performance and to reduce memory consumption.
Many thanks in advance!!!

**Comment by vnbaaij** [comment, marked as answer]
Hi,

Internally we are re-using the QuickGrid code. Everything mentioned [here](https://aspnet.github.io/quickgridsamples/datasources) in the 'Entity Framework IQueryable data' section is applicable to our grid as well. See also https://www.fluentui-blazor.net/#use-the-datagrid-component-with-ef-core

Oh, and remember to install the [DataGrid EF Adpater package](https://www.nuget.org/packages/Microsoft.FluentUI.AspNetCore.Components.DataGrid.EntityFrameworkAdapter)
